File: hackerrank/practice/solutions/src/solutions/timeConversion.java
package solutions;
import java.io.*;
public class timeConversion {

	public static void main(String args[])throws IOException{
		BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
		String time  = input.readLine();
		String flag = time.substring(8);
		String hour;
		int h1 = Integer.parseInt(time.substring(0, 2));
		if (flag.equals("PM")){
			if (h1 != 12){
				h1 += 12;
			}
			hour = String.valueOf(h1);
		}
		else{
			if (h1 == 12) hour = "00";
			else{
				hour = time.substring(0,2);
			}
		}
		String ntime = hour + time.substring(2, 8);
		System.out.println(ntime);
	}
}

File: hackerrank/practice/archive/time-conversion.py
time = raw_input()
flag = time[-2:]
if flag == "PM":
	if time[:2] != "12":
		hour = str(int(time[:2])+12)
	else: hour = "12"
	#print "Hour1: " , hour
if flag == "AM":
	if time[:2] == "12":
		hour = "00"
	else:
		hour = time[:2]
	#print "Hour2: " , hour
print hour + time[2:8]